src/abbfreeathome/devices/switch_sensor.py

```diff
@@ -92,12 +112,119 @@ def _refresh_state_from_datapoint(self, datapoint: dict[str, Any]) -> str:
 
             self._state = self._switch_sensor_state
             return "state"
+        if datapoint.get("pairingID") == Pairing.AL_INFO_ON_OFF.value:
+            try:
+                _value = int(
+                    self.get_device_parameter(
+                        parameter=Parameter.PID_LED_OPERATION_MODE
+                    )[1]
+                )
+            except InvalidDeviceChannelParameter:
+                return None
+
+            if _value == 2:
+                self._led = datapoint.get("value") == "1"
+                return "led"
```

Comment:
Although nothing technically wrong with this implementation, here's a way to simplify things a bit. No need to convert the value to an int, and set a named variable from the tuple return to let the function unpack the list of values.

We can also wrap the whole thing in the try/catch.

```suggestion
        if datapoint.get("pairingID") == Pairing.AL_INFO_ON_OFF.value:
            try:
                _, _parameter_value = self.get_device_parameter(
                    parameter=Parameter.PID_LED_OPERATION_MODE
                )
                if _parameter_value == "2":
                    self._led = datapoint.get("value") == "1"
                    return "led"
            except InvalidDeviceChannelParameter:
                return None
```